File: Wander/Assets/Networking/Response/ResponseLobbies.cs
using UnityEngine;

using System;

public class ResponseLobbiesEventArgs : ExtendedEventArgs {

	public short status { get; set; }
	public string json { get; set; }
	public ResponseLobbiesEventArgs() {
		event_id = Constants.SMSG_LOBBY;
	}
}

public class ResponseLobbies : NetworkResponse {
	
	private short status;
	private string json;

	public ResponseLobbies() {

	}
	
	public override void parse() {
		status = DataReader.ReadShort(dataStream);
		if (status == 0) {
			json = DataReader.ReadString(dataStream);
		}
	}
	
	public override ExtendedEventArgs process() {
		ResponseLobbiesEventArgs args = null;
		if(status == 0){
			args = new ResponseLobbiesEventArgs();
			args.status = status;
			args.json = json;
		} 
		return args;
	}
}

File: Wander/WoB_Server_Basic/src/networking/response/ResponseLobbies.java
package networking.response;

// Other Imports
import metadata.Constants;
import utility.GamePacket;

/**
 * The ResponseLogin class contains information about the authentication
 * process.
 */
public class ResponseLobbies extends GameResponse {

    private short status;
    private String json;

    public ResponseLobbies() {
        responseCode = Constants.SMSG_LOBBY;
    }

    @Override
    public byte[] constructResponseInBytes() {
        GamePacket packet = new GamePacket(responseCode);
        packet.addShort16(status);
        packet.addString(json);        
        return packet.getBytes();
    }

    public void setStatus(short status){
        this.status = status;
    }

    public void setJson(String json){
        this.json = json;
    }

}